bob-backend/src/pearl/group.rs

```diff
@@ -569,39 +598,72 @@ impl Group {
         });
     }
 
-    pub async fn delete(&self, key: BobKey) -> Result<u64, Error> {
-        let holders = self.holders.read().await;
+    pub async fn delete(
+        &self,
+        key: BobKey,
+        timestamp_config: StartTimestampConfig,
+        force_delete: bool,
+    ) -> Result<u64, Error> {
+        {
+            let holders = self.holders.read().await;
+            if holders.len() != 0 {
+                let copy = holders.iter().cloned().enumerate().collect();
+                drop(holders);
+                return self.delete_in_holders(copy, key, force_delete).await;
```

Comment:
Do this under the `lock` directly on `holders` collection. To avoid deadlock it is better to acquire `write` lock above

Reply:
After https://github.com/qoollo/pearl/pull/219 will be merged, we can start to use `read` lock here